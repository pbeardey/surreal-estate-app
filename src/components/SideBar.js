import React from "react";
import { Link } from "react-router-dom";
import "../styles/SideBar.css";

const SideBar = () => {
  return (
    <div className="side-bar">
      <ul>
        <li>
          <Link to="/">All</Link>
        </li>
        <li>
          <Link to={`/?query={"city": "Manchester"}`}>Manchester</Link>
        </li>
        <li>
          <Link to={`/?query={"city": "Leeds"}`}>Leeds</Link>
        </li>
        <li>
          <Link to={`/?query={"city": "Liverpool"}`}>Liverpool</Link>
        </li>
        <li>
          <Link to={`/?query={"city": "Sheffield"}`}>Sheffield</Link>
        </li>
      </ul>
    </div>
  );
};

export default SideBar;
