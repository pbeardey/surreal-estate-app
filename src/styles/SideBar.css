.side-bar {
  background-color: lightgoldenrodyellow;
}

.side-bar > ul {
  width: 200px;
  margin: 2px;
  padding-inline-start: 0px;
  padding: 2px;
  text-align: center;
  list-style: none;
}

.side-bar > ul > li {
  border-style: solid;
  border-width: 2px;
  border-color: lightgray;
  border-radius: 5px;
  margin: 2px;
}

.side-bar > ul > li :hover {
  background-color: goldenrod;

}