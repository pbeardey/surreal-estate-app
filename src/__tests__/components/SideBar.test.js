import React, { render, screen } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import SideBar from "../../components/SideBar";

describe("Sidebar renders", () => {
  it("renders SideBar", () => {
    const { asFragment } = render(
      <Router>
        <SideBar />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders 5 options", () => {
    render(
      <Router>
        <SideBar />
      </Router>
    );
    const listItems = screen.getAllByRole("listitem");

    expect(listItems).toHaveLength(5);
  });
});
