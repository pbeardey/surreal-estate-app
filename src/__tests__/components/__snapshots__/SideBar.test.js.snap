// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders renders SideBar 1`] = `
<DocumentFragment>
  <div
    class="side-bar"
  >
    <ul>
      <li>
        <a
          href="/"
        >
          All
        </a>
      </li>
      <li>
        <a
          href="/?query={\\"city\\": \\"Manchester\\"}"
        >
          Manchester
        </a>
      </li>
      <li>
        <a
          href="/?query={\\"city\\": \\"Leeds\\"}"
        >
          Leeds
        </a>
      </li>
      <li>
        <a
          href="/?query={\\"city\\": \\"Liverpool\\"}"
        >
          Liverpool
        </a>
      </li>
      <li>
        <a
          href="/?query={\\"city\\": \\"Sheffield\\"}"
        >
          Sheffield
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
